tests/inverted-index-testSpec.js

```diff
@@ -1,89 +1,262 @@
+        .attach('book', './fixtures/valid.json')
+        .attach('book', './fixtures/anothervalid.json')
+        .end((err, res) => {
+          expect(res.status).toEqual(200);
+          done(err);
+        });
+  });
+
+  it('should create an index when valid file is uploaded ', (done) => {
+    api.post('/api/v0/create')
+     .attach('book', './fixtures/valid.json')
+     .attach('book', './fixtures/anothervalid.json')
+        .end((err, res) => {
+          const result = {
+            'valid.json': {
+              when: [0, 1, 2],
+              we: [0, 1, 2],
+              were: [0, 1, 2],
+              boys: [0],
+              guys: [1, 2]
+            },
+            'anothervalid.json': {
+              when: [0, 2],
+              we: [0, 2],
+              were: [0, 2],
+              boys: [0],
+              am: [1],
+              alice: [1],
+              girls: [2]
+            }
+          };
+          console.log(JSON.parse(res.text));
+          expect(JSON.parse(res.text)).toEqual(result);
+          done();
+        });
+  });
+
+  it('responds with the right status code for single file', (done) => {
+    api.post('/api/v0/create')
+        .attach('book', './fixtures/valid.json')
+        .end((err, res) => {
+          expect(res.status).toEqual(200);
+          done(err);
+        });
+  });
+
+  it('should return search result when fileName argument is not supplied', (done) => {
+    api.post('/api/v0/create')
+      .attach('book', './fixtures/anothervalid.json')
+      .attach('book', './fixtures/valid.json')
+        .end(() => {
+          api.post('/api/v0/search')
+        .send({
+          terms: ['boys', 'we', 'when']
+        })
+        .end((err, res) => {
+          expect(res.status).toEqual(200);
+          done(err);
+        });
+        });
+  });
+
+  it('should return result for search when fileName argument is supplied', (done) => {
+    api.post('/api/v0/create')
+        .attach('book', './fixtures/valid.json')
+       .attach('book', './fixtures/anothervalid.json')
+        .end(() => {
+          api.post('/api/v0/search')
+          .send({
+            searchTerms: ['we', 'when'],
+            fileName: 'valid.json'
+          })
+          .end((err, res) => {
+            expect(res.status).toEqual(200);
+            done(err);
+          });
+        });
+  });
+
+  it('type of created index should be object', (done) => {
+    api.post('/api/v0/create')
+      .attach('book', './fixtures/valid.json')
+      .end((err, res) => {
+        expect(typeof res.body).toEqual('object');
+        done();
+      });
+  });
+
+  it('should give an Error message when empty file is uploaded', (done) => {
+    api.post('/api/v0/create')
+   .attach('book', './fixtures/empty.json')
+      .end((err, res) => {
+        expect(res.text).toEqual('Book is Empty');
+        done();
+      });
+  });
+
+  it('should give an Error message when Malformed is uploaded ', (done) => {
```

Comment:
Good work. I noticed you finally found a way to ascertain that the correct error message is generated when testing with supertest, and that's commendable. Your code is also readable. Good job!